test/trackForMutations.spec.js

```diff
@@ -238,4 +264,67 @@ describe('trackForMutations', () => {
       testCasesForNonMutation(nonMutations[nonMutationDesc]);
     });
   });
+
+  Object.keys(nonMutations).forEach((nonMutationDesc) => {
+    describe(`(ignore) ${nonMutationDesc}`, () => {
+      testCasesForIgnoreMutationDetectionByPath(nonMutations[nonMutationDesc]);
+    });
+  });
+
+  it('can ignore nested branches from mutation detection', () => {
```

Comment:
Same idea as above: let's move all the cases into the mutation and non mutations groups, and specify the middleware options in the cases where necessary.